src/cr/cube/cube.py

```diff
@@ -457,36 +490,34 @@ def _is_single_filter_col_cube(self):
         return self._cube_dict["result"].get("is_single_col_cube", False)
 
     @lazyproperty
-    def _mean_references(self):
-        """Dict of mean references, tipically for numeric arrays."""
+    def _numeric_measure_references(self):
+        """Dict of numeric measure references, tipically for numeric means or sum."""
+        if not self._numeric_measures:
+            return {}
         cube_response = self._cube_response
         cube_measures = cube_response.get("result", {}).get("measures", {})
-        metadata = cube_measures.get("mean", {}).get("metadata", {})
+        metadata = cube_measures.get(self._numeric_measures[0], {}).get("metadata", {})
         return metadata.get("references", {})
 
     @lazyproperty
-    def _mean_subvariables(self):
+    def _numeric_measure_subvariables(self):
         """List of mean subvariables, tipically for numeric arrays."""
+        if not self._numeric_measures:
+            return []
         cube_response = self._cube_response
         cube_measures = cube_response.get("result", {}).get("measures", {})
-        metadata = cube_measures.get("mean", {}).get("metadata", {})
+        metadata = cube_measures.get(self._numeric_measures[0], {}).get("metadata", {})
         return metadata.get("type", {}).get("subvariables", [])
 
     @lazyproperty
-    def _measure(self):
-        """_BaseMeasure subclass representing primary measure for this cube.
+    def _numeric_measures(self):
+        """List of numeric measures expressed in the cube_response.
 
-        If the cube response includes a means measure, the return value is
-        means. Otherwise it is counts, with the choice between weighted or
-        unweighted determined by whether the cube has weighted counts.
+        Basically the numeric measures are the intersection between all the measures
+        within the cube response and the defined QUANTITY_OF_INTEREST_MEASURES.
         """
-        return (
-            self._measures.means
-            if self._measures.means is not None
-            else self._measures.weighted_counts
-            if self._measures.weighted_counts is not None
-            else self._measures.unweighted_counts
-        )
+        numeric_measures = [measure.value for measure in NUMERIC_MEASURES]
+        return list(self.available_measures.intersection(numeric_measures))
```

Comment:
As mentioned a few times, we should not be trafficking in str values. The type of this should be a collection of MEASURE enum members.

Also, this should be immutable. No @lazyproperty should be mutable, otherwise it can return different values on different calls. `frozenset` may be convenient, that has ready set operations and can be iterated, it just can't be directly indexed, so you would need `list(self._available_numeric_measures)[0]` to get a random one, like up above in `._numeric_mean_subvariables`.

Also, I think this should be `._available_numeric_measures`, to be more descriptive. It is not _all_ numeric values, just the ones available in this cube-result.

The `QUANTITY_OF_INTEREST_MEASURES` reference is outdated.

If you need the str value elsewhere, like up on line 509, you can just use the `.value` property on the enum member, so that line would become:

```python
metadata = cube_measures.get(self._available_numeric_measures[0].value, {}).get("metadata", {})
```